packages/core/src/agent/index.ts

```diff
@@ -1376,7 +1376,7 @@ export class Agent<
               runId,
             });
           }
-          void onFinish?.(result);
+          await Promise.resolve(onFinish?.(result));
```

Comment:
```suggestion
          await onFinish?.(result);
```

Reply:
That's what I had initially and @abhiaiyer91 asked me to use `Promise.resolve` see his comment above.

Reply:
Thats my bad! I have a habit of using it to make it very explicit but its not really needed!

Reply:
No worries. I've made the updates